NeuroID/NeuroIDClass/Extensions/NIDUser.swift

```diff
@@ -145,4 +145,21 @@ public extension NeuroID {
 
         return res
     }
+
+    /**
+        This should be called the moment a user trys to login. Returns true or false based on if a passed in UserID is valid.
+        @param {String} [attemptedRegisteredUserId] - an optional identifier for the login
+     */
+    static func attemptedLogin(_ attemptedRegisteredUserId:String? = nil) -> Bool{
+        if (attemptedRegisteredUserId != nil) {
```

Comment:
If the value is nil or fails validation we need to set the UID value like [trackjs does here LN 1551](https://github.com/Neuro-ID/trackjs/commit/d3eef8c3894b2e831f3ebf1e48fba168380ae363)